test/lit.cfg

```diff
@@ -930,39 +930,68 @@ elif run_os in ['linux-gnu', 'linux-gnueabihf', 'freebsd', 'windows-cygnus', 'wi
         "clang++ -target %s %s" %
         (config.variant_triple, clang_mcp_opt))
     config.target_ld = "ld -L%r" % (make_path(test_resource_dir, config.target_sdk_name))
-elif run_os == 'linux-androideabi':
+elif run_os == 'linux-androideabi' or run_os == 'linux-android':
+    def get_architecture_value(**kwargs):
+        result = kwargs[run_cpu]
+        if result is None:
+          if run_cpu == "armv7s" or run_cpu == "armv7k":
+            result = kwargs["armv7"]
```

Comment:
I would rewrite this as: `if run_cpu.startswith('armv7'): return kwargs["armv7"]`